backend/cfg/cfg_dataflow.ml

```diff
@@ -321,10 +321,12 @@ module Backward
                 := WorkSet.filter
                      (fun { WorkSetElement.label; value } ->
                        if Label.equal label predecessor_label
-                       then (
+                       then
                          if D.less_equal new_value value
-                         then already_in_workset := true;
-                         not (D.less_equal value new_value))
+                         then (
+                           already_in_workset := true;
+                           true)
+                         else false
                        else true)
                      !work_set;
               if not !already_in_workset
```

Comment:
Is it possible for `already_in_workset` be `true` here in Line 330 (of the original code)? Note that `new_value` is strictly the biggest we observed so far. If so, L323-328 could just be replaced with `not (Label.equal label predecessor_label)`, possibly preceeded by an `assert (not (D.less_equal new_value value))`.

Reply:
I think you're right.
But it's really confusing because we have value in `res_block` and in `work_set` but we enforce it to be the same.
Do you know why even `value` in `WorkSetElement.t` even exists?

Reply:
yes, looks like in the current version, it'd be enough to keep the label of the block in the `work_set`. 

iirc, a previous `work_set` implementation could have more than one value per block.